ts/packages/node/src/stores/LaunchDarkly.ts

```diff
@@ -0,0 +1,173 @@
+        },
+      ],
+    };
+    await this.patchUpdate(config, updateDefaultVariationData);
+    return this.patchUpdate(config, updateFallthroughVariationData);
+  }
+
+  private async upsert(config: Config) {
+    if (!this.environments.includes(config.set)) {
+      await this.createEnvironment(config);
+    }
+    let featureFlag;
+    try {
+      featureFlag = await this.getFeatureFlag(config);
+      const existingVariation =
+        featureFlag.variations.find((variation: any) => variation.value === config.value) ||
+        (await this.createVariation(config));
+      if (existingVariation) featureFlag = await this.updateDefaultFallthroughVariation(config, existingVariation._id);
+    } catch (e) {
+      if (e.response.status === 404) featureFlag = await this.createFeatureFlag(config);
+      else throw e;
+    }
+    if (featureFlag && !featureFlag.environments[config.set].on) {
+      const onData = {
+        environmentKey: config.set,
+        instructions: [{ kind: 'turnFlagOn' }],
+      };
+      await this.patchUpdate(config, onData);
+    }
+  }
+
+  private async validateQueries(queries: ConfigStoreQuery[]): Promise<[string, string[]]> {
+    await this.updateEnvironments();
+    queries.forEach((query) => {
+      assert(query.set.split('/').length === 1, 'Subsets are not supported in LaunchDarklyStore');
+    });
+    const noRootQueries = queries.filter((query) => query.set !== '' && this.environments.includes(query.set));
+    if (noRootQueries[0]) {
+      const env = noRootQueries[0].set;
+      const keys = noRootQueries.map((value) => value.key);
+      return [env, keys];
+    }
+    throw new Error('No valid queries');
+  }
+
+  private assertNotSettingToRoot(configs: Config[]) {
+    assert.strictEqual(
+      configs.some((config) => config.set === ''),
+      false,
+      'ConfigSet cannot be empty string',
+    );
+  }
+
+  private async getEnvFeatureFlags(env: string, keys: string[]): Promise<Config[]> {
+    const { data: featureFlags } = await this.client.get(`/flags/${this.projectKey}?env=${env}`);
+    return featureFlags.items
+      .filter((featureFlag: any) => keys.includes(featureFlag.key))
+      .map((featureFlag: any) => {
+        const onValue = Object.entries(featureFlag.environments[env]._summary.variations)
+          .filter((value: [string, any]) => value[1].isFallthrough)
+          .map((value) => featureFlag.variations[parseInt(value[0], 10)].value)[0];
+        return { key: featureFlag.key, set: env, value: featureFlag.environments[env].on ? onValue : '' };
+      });
+  }
+
+  async get(queries: ConfigStoreQuery[]): Promise<Config[]> {
+    try {
+      const [env, keys] = await this.validateQueries(queries);
+      return this.getEnvFeatureFlags(env, keys);
+    } catch (e) {
+      return [];
+    }
+  }
+
+  async set(configs: Config[]): Promise<void> {
+    this.assertNotSettingToRoot(configs);
+    await this.updateEnvironments();
+    configs.forEach(this.upsert.bind(this));
```

Comment:
Is there a reason why `promise.all` isn't used here?

Reply:
Maybe i'm missing something but it exactly the same... 
i'll use promise.all

how about this?
```javascript
const upsertPromises = configs.map(this.upsert.bind(this));
await Promise.all(upsertPromises);
```

if you'd decide that I shouldn't keep the envs in mem then i'll have to create them all first.